CRM/Admin/Form/MailSettings.php

```diff
@@ -94,6 +95,26 @@ public function addRules() {
     $this->addFormRule(array('CRM_Admin_Form_MailSettings', 'formRule'));
   }
 
+  public function getDefaultEntity() {
+    return 'MailSettings';
+  }
+
+  /**
+   * Add local and global form rules.
+   */
+  public function setDefaultValues() {
+    $defaults = parent::setDefaultValues();
+
+    // Set activity status to "Completed" by default.
+    if ($this->_action != CRM_Core_Action::DELETE &&
+      (!$this->_id || !CRM_Core_DAO::getFieldValue('CRM_Core_BAO_MailSettings', $this->_id, 'activity_status'))
+    ) {
+      $defaults['activity_status'] = 'Completed';
```

Comment:
Just to check if I understand this conditional:

IF (this is not a deletion) AND ((this is a creation) OR (there is no default value for activity status)) THEN status should be "Completed"

Does that sound right?